src/firefly/js/templates/hydra/HydraViewer.jsx

```diff
@@ -152,39 +153,89 @@ function closeExpanded() {
-    setIfUndefined(mSlotProps,'topSection.component', Greetings);      // use custom topSection
-    setIfUndefined(mSlotProps,'contentSection.sx', {maxWidth: '80em', mx: 'auto'});   // limit page's width
+const HydraAppBranding = ({title, desc}) => (
+    <Stacker direction='column' alignItems='start'>
+        <Typography level='h4'>{title || 'Welcome message here'}</Typography>
+        <Typography level='body-md'>{desc || 'Additional description of this application'}</Typography>
+    </Stacker>
+);
+
+export function HydraLanding({icon, title, desc, bgImage, slotProps={}, ...props} ) {
+    const mSlotProps = cloneDeep(slotProps);
+    defaultsDeep(mSlotProps, {
+        bgMonitorHint: { sx: { right: 50 } },
+        topSection: { title, desc }
+    });
+
+    if (bgImage) {
+        // visually combine topSection & bottomSection into contentSection that can contrast with the bgContainer
+        defaultsDeep(mSlotProps, {
+            bgContainer: {
+                sx: {
+                    display: 'flex', flexGrow: 1,
+                    backgroundImage: `url(${bgImage})`, backgroundSize: 'cover', backgroundPosition: 'center'
+                }
+            },
+            contentSection: {
+                sx: (theme) => {
+                    // background image will remain same in both the theme modes, and we need text to contrast with the image,
+                    // so we create a translucent dark overlay and put text over it as if it's dark theme
+                    const darkPalette = theme.colorSchemes.dark.palette;
+                    return {
+                        maxWidth: '56rem', m: 'auto',
+                        backgroundColor: hexColorWithAlpha(darkPalette.background.surface.split(', ')?.[1]?.slice(0, -1) ?? '#000', 0.6),
+                        backdropFilter: 'blur(1px)',
+                        '.MuiTypography-body-md, .MuiTypography-body-lg': { color: darkPalette.text.secondary },
+                        '.MuiTypography-h2': { color: darkPalette.text.primary },
+                        '.MuiTypography-colorPrimary': { color: `rgb(${darkPalette.primary.mainChannel})` },
+                        '.MuiDivider-root': { backgroundColor: darkPalette.neutral.solidBg }
+                    };
+                },
+                divider: <Divider sx={{ width: '4rem', alignSelf: 'center' }} />
+            },
+            bottomSection: {
+                icon: false, // to prevent default being applied (on undefined)
+                slotProps: {
+                    root: { sx: { py: 4, pb: 0, backgroundColor: 'transparent' } }
+                }
+            }
+        });
+    }
+    else {
+        defaultsDeep(mSlotProps, {
+            contentSection: { sx: { maxWidth: '80rem', mx: 'auto' } },
+            topSection: { component: HydraAppBranding }, // use custom topSection instead of DefaultAppBranding
+            bottomSection: { icon }
+        });
+    }
```

Comment:
I see what you're doing now. Would it improve readability to define `defaultProps` and `defaultPropsWithBgImage` separately? Then, apply `defaultsDeep` to the appropriate defaults based on `bgImage`. Just a suggestion.

Reply:
Good idea, done! The logic is more functional now